backend/sftp/sftp.go

```diff
@@ -389,6 +294,10 @@ func (r *SFTP) Create() (backend.Blob, error) {
 
 // Construct path for given backend.Type and name.
 func (r *SFTP) filename(t backend.Type, name string) string {
```

Comment:
hm this function looks identical to the one in `local.go`? the `dirname` function also looks similar.. why is this code duplicated and not shared by both backends?